wasm/rollup.config.js

```diff
@@ -1,28 +1,64 @@
 import rust from "@wasm-tool/rollup-plugin-rust";
 
-export default {
-    input: {
-        index: "./js/index.js",
-        worker: "./js/worker.js",
+// This uses 3 separate builds, instead of 1 build.
+//
+// The reason is because the `worker.js` file needs to be
+// fully self-contained, it cannot contain imports to other
+// chunks.
+//
+// But Rollup doesn't support standalone entry points, so we
+// hack around it by instead using multiple builds.
+//
+// But we want to share the Wasm build between the `index.js`
+// and `worker.js` builds, so we build the Wasm, and then
+// build the `index.js` and `worker.js` separately.
```

Comment:
It does increase the file size a little bit for the Worker, because it has to include the JS shims twice. But the JS shim code is relatively small (36.7 kilobytes). So it only increases the file size by 36.7 kilobytes.

It does not affect the Wasm size at all, there is 1 Wasm which is shared between the main thread and the Workers.